statesync/block.go

```diff
@@ -0,0 +1,59 @@
+// Copyright (C) 2024, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package statesync
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/ava-labs/avalanchego/ids"
+	"github.com/ava-labs/avalanchego/snow/engine/snowman/block"
+)
+
+var _ block.StateSummary = (*SyncableBlock[StateSummaryContainer])(nil)
+
+type StateSummaryContainer interface {
+	ID() ids.ID
+	Height() uint64
+	Bytes() []byte
+	GetStateRoot() ids.ID
+	MarkAccepted(context.Context)
+	fmt.Stringer
+}
+
+type SyncableBlock[T StateSummaryContainer] struct {
+	container T
+	accepter  Accepter[T] // accepter is nil if the SyncableBlock is constructed by the server
```

Comment:
Another idea instead of relying on not ever calling `Accept` to avoid panic'ing would be to just pass in a no-op implementation of accept so you don't have to document this edge-case at all.

Reply:
We should never call Accept on the server side, so I think we should log an error rather than pass a no-op if we change this.